### PR TITLE
[FLINK-29741] Remove flink-scala and flink-streaming-scala dependency exclude table module

### PR DESCRIPTION
**Pull request by codenohup**
## What is the purpose of the change

Remove the flink-scala and flink-streaming-scala dependencies, excluding the table module, to facilitate the removal of deprecated Scala APIs.

## Brief change log
 - delete flink-cep-scala module
 - remove scala api test in flink-hadoop-compatibility module
 - remove flink-scala and flink-streaming-scala dependency in flink-avro, flink-connector-hive, flink-tests modules

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-tests/src/test/scala/org/apache/flink/api/scala/functions/ClosureCleanerITCase.scala [hunk 1]
```diff
@@ -1,262 +0,0 @@
...
-import org.junit.{After, Before, Rule, Test}
-import org.junit.Assert.fail
-import org.junit.rules.TemporaryFolder
-import org.junit.runner.RunWith
-import org.junit.runners.Parameterized
-
-/* The test cases are originally from the Apache Spark project. Like the ClosureCleaner itself. */
-
-@RunWith(classOf[Parameterized])
-class ClosureCleanerITCase(mode: TestExecutionMode) extends MultipleProgramsTestBaseJUnit4(mode) {
```
Comment (reswqa) [inline flink-tests/src/test/scala/org/apache/flink/api/scala/functions/ClosureCleanerITCase.scala:36]: Can not remove this.

File: flink-tests/src/test/scala/org/apache/flink/api/scala/migration/StatefulJobSavepointMigrationITCase.scala [hunk 2]
```diff
@@ -1,380 +0,0 @@
...
-
-import javax.annotation.Nullable
-
-import java.util
-import java.util.function.BiFunction
-import java.util.stream.Collectors
-
-import scala.util.{Failure, Try}
-
-object StatefulJobSavepointMigrationITCase {
```
Comment (reswqa) [inline flink-tests/src/test/scala/org/apache/flink/api/scala/migration/StatefulJobSavepointMigrationITCase.scala:57]: Can not remove tests from `migration` dir.

File: flink-formats/flink-avro/src/test/resources/flink_11-kryo_registrations [hunk 3]
```diff
@@ -0,0 +1,86 @@
+0,int
```
Comment (reswqa) [inline flink-formats/flink-avro/src/test/resources/flink_11-kryo_registrations:1]: Why we need this file?

File: flink-dist-scala/pom.xml [hunk 4]
```diff
@@ -127,4 +127,4 @@ under the License.
 			</plugin>
 		</plugins>
 	</build>
-</project>
\ No newline at end of file
+</project>
```
Comment (reswqa) [inline flink-dist-scala/pom.xml:130]: ?

File: flink-connectors/flink-hadoop-compatibility/src/test/scala/org/apache/flink/api/hadoopcompatibility/scala/WordCountMapredITCase.scala [hunk 5]
```diff
@@ -1,89 +0,0 @@
-/*
```
Comment (codenohup) [inline flink-connectors/flink-hadoop-compatibility/src/test/scala/org/apache/flink/api/hadoopcompatibility/scala/WordCountMapredITCase.scala:1]: The `WordCountMapredITCase` has a corresponding Java version, so delete it directly.

File: flink-connectors/flink-hadoop-compatibility/src/test/scala/org/apache/flink/api/hadoopcompatibility/scala/WordCountMapreduceITCase.scala [hunk 6]
```diff
@@ -1,91 +0,0 @@
-/*
```
Comment (codenohup) [inline flink-connectors/flink-hadoop-compatibility/src/test/scala/org/apache/flink/api/hadoopcompatibility/scala/WordCountMapreduceITCase.scala:1]: The WordCountMapreduceITCase has a corresponding Java version, so delete it directly.

File: flink-tests/src/test/scala/org/apache/flink/api/scala/migration/StatefulJobSavepointMigrationITCase.scala (hunk 2 above)
Comment (codenohup) [reply flink-tests/src/test/scala/org/apache/flink/api/scala/migration/StatefulJobSavepointMigrationITCase.scala:57]: This test case primarily focus on Scala case class state migration, while the corresponding Java version is `org.apache.flink.test.checkpointing.StatefulJobSnapshotMigrationITCase`. Therefore, I believe this test case can be deleted. What do you think?

File: flink-tests/src/test/scala/org/apache/flink/api/scala/functions/ClosureCleanerITCase.scala (hunk 1 above)
Comment (codenohup) [reply flink-tests/src/test/scala/org/apache/flink/api/scala/functions/ClosureCleanerITCase.scala:36]: I have rewritten it in Java. You can find it in `org.apache.flink.api.functions.ClosureCleanerITCase`.

**Pull request merged**